framework/db/ActiveQuery.php

```diff
@@ -790,6 +790,29 @@ public function alias($alias)
         return $this;
     }
 
+
+    /**
+     * Returns conditions with alias
+     * @param array $conditions array of condition, keys to be modified
+     * @param null|string $alias set empty string for no apply alias. Set null for apply primary table alias
+     * @return array
+     */
+    public function applyTableAlias($conditions, $alias = null)
```

Comment:
I don't think it's a good idea to have it as a public method. I'd rather move it to private methods in unique and exist validators even if it duplicates code a bit.